### PR TITLE
Scala code in container-di converted to Java

### PR DESCRIPTION
**Pull request by ollivir**
Includes some changes in other modules where they interfaced with container-di using Scala container classes, or where the removal of scala Maven artifacts revealed an import of scala classes otherwise.

In the tests, some code was shuffled between ContainerTest and ContainerTestBase to avoid running tests twice without any change in other classes that inherited from them.

CC: @gjoranv

### CODE REVIEW

File: container-di/src/main/java/com/yahoo/container/di/Container.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    // Reload config manually, when subscribing to non-configserver sources
+    public void reloadConfig(long generation) {
+        subscriberFactory.reloadActiveSubscribers(generation);
+    }
+
+    private void deconstructAllComponents(ComponentGraph graph, ComponentDeconstructor deconstructor) {
+        graph.allComponentsAndProviders().forEach(deconstructor::deconstruct);
+    }
+
+    private static final Logger log = Logger.getLogger(Container.class.getName());
```
Comment (bjorncs) [inline container-di/src/main/java/com/yahoo/container/di/Container.java]: Consider moving the field declaration to top of file.

**Review by bjorncs — APPROVED** [review]
LGTM

**Comment by bjorncs** [comment]
@gjoranv Please review as well.

File: container-di/src/main/java/com/yahoo/container/di/Container.java (hunk 1 above)
Comment (ollivir) [reply container-di/src/main/java/com/yahoo/container/di/Container.java]: Ahh, missed that one. Fixed.

**Review by gjoranv — APPROVED** [review]
👍

**Pull request merged**